lxd/container.go

```diff
@@ -836,7 +836,15 @@ func instanceCreateInternal(s *state.State, args db.InstanceArgs) (instance.Inst
 	}
 
 	if args.Profiles == nil {
-		args.Profiles = []string{"default"}
+		if args.BaseImage == "" {
+			args.Profiles = []string{"default"}
+		} else {
+			profiles, err := s.Cluster.ImageProfilesGet(args.BaseImage, args.Project)
+			if err != nil {
+				return nil, err
+			}
+			args.Profiles = profiles
+		}
```

Comment:
This isn't the right location for this change. `BaseImage` is also used with operations like migration and your change will cause failures in some cases there.

Instead you want the image profile logic to be done in `instanceCreateFromImage`.